Py_FS/wrapper/nature_inspired/SSA.py

```diff
@@ -0,0 +1,180 @@
+Swarm and evolutionary computation, 44, 148-175."
+
+"""
+
+import math
+
+import numpy as np
+from sklearn import datasets
+
+from Py_FS.wrapper.nature_inspired._utilities_test import initialize as initialize_
+from Py_FS.wrapper.nature_inspired._utilities_test import sort_agents as sort_agents_
+from Py_FS.wrapper.nature_inspired.algorithm import Algorithm
+
+
+class SSA(Algorithm):
+
+    # Squirrel Search Algorithm
+    ############################### Parameters ####################################
+    #                                                                             #
+    #   num_agents: number of squirrels                                           #
+    #   max_iter: maximum number of generations                                   #
+    #   train_data: training samples of data                                      #
+    #   train_label: class labels for the training samples                        #
+    #   obj_function: the function to maximize while doing feature selection      #
+    #   trans_function_shape: shape of the transfer function used                 #
+    #   save_conv_graph: boolean value for saving convergence graph               #
+    #                                                                             #
+    ###############################################################################
+
+    def __init__(self,
+                 num_agents,
+                 max_iter,
+                 train_data,
+                 train_label,
+                 save_conv_graph=False,
+                 trans_func_shape='s',
+                 seed=0):
+
+        super().__init__(num_agents=num_agents,
+                         max_iter=max_iter,
+                         train_data=train_data,
+                         train_label=train_label,
+                         save_conv_graph=save_conv_graph,
+                         trans_func_shape=trans_func_shape,
+                         seed=seed)
+
+        self.algo_name = 'SSA'
+        self.agent_name = 'Squirrel'
+        self.algo_params = {}
+
+    def user_input(self):
+        # initializing parameters
+        pdp = self.algo_params['pdp'] = float(input('Predator Presence Probability: ') or 0.1)
+        row = self.algo_params['row'] = float(input('Density of air: ') or 1.204)
+        V = self.algo_params['V'] = float(input('Speed: ') or 5.25)
+        S = self.algo_params['S'] = float(input('Surface area of body: ') or 0.0154)
+        cd = self.algo_params['cd'] = float(input('Frictional Drag Coefficient: ') or 0.6)
+        CL = self.algo_params['CL'] = float(input('Lift Coefficient: ') or 0.7)
+        hg = self.algo_params['hg'] = float(input('Loss in height occurred after gliding: ') or 1)
+        sf = self.algo_params['sf'] = float(input('Scaling Factor: ') or 18)
+        Gc = self.algo_params['Gc'] = float(input('Gliding Constant: ') or 1.9)

```

Comment:
Minor comment:
As a new user, it would be better if we include the approximate range of a variable within the input. For example, you could say float(input('Predator Presence Probability [0-1] :') or 0.1). For other variables, please provide a range to the best of your knowledge.